tests/test_collection.py

```diff
@@ -33,12 +33,15 @@ def test_collection_not_exists(client):
     assert response.status_code == 404
 
 
-def test_collection_not_published_404(admin_client, client, draft_collection):
+def test_collection_not_yet_published_returns_404(
+    admin_client, client, draft_collection):
     url_kwargs = {'pk': draft_collection.pk, 'slug': draft_collection.slug}
     url = reverse('product:collection', kwargs=url_kwargs)
     response = client.get(url)
     assert response.status_code == 404
 
+    # A non staff user should not have access to collections yet to be published
+    # A staff user should have access to collections yet to be published
```

Comment:
I meant above `client.get(...)` and `admin_client.get(...)`, that would make more sense.